room favorite

pyclan88 opened the pull request:
Спринт 21. Избранные треки

pyclan88 committed:
room favorite

IliaAnK commented on app/src/main/java/com/practicum/playlistmaker/search/data/db/dao/FavoriteTrackDao.kt:
```diff
@@ -0,0 +1,28 @@
+    suspend fun deleteTrack(favoriteTrackEntity: FavoriteTrackEntity)
+
+    @Query("SELECT * FROM favorite_track_table ORDER BY addedAt DESC")
+    suspend fun getAllTracks(): List<FavoriteTrackEntity>
```
Рекомендация:
`Room` может возвращать `Flow`
Получится:
`suspend fun getAllTracks(): Flow<List<FavoriteTrackEntity>>`

Аналогично для других мест, где требуется возвращать `Flow`

pyclan88 replied:
если Flow добавляем, то от suspend избавляемся, иначе вылетает ошибка:
Dao functions that have a suspend modifier must not return a deferred/async type (kotlinx.coroutines.flow.Flow). Most probably this is an error. Consider changing the return type or removing the suspend modifier.

IliaAnK commented:
Рекомендация:
Папку `.idea` стоит добавить в `gitignore` файл. И удалить её из гитхаб репозитория

pyclan88 committed:
room favorite
повторная сдача работы с исправлениями:
критические:
1. исправлено отображение разделителя на BottomNavBar
2. исправлено отображение "Вы искали" на экране "Поиск" при отображении истории
3. исправлен баг с отображением информации при возвращении на экран "Поиск"

pyclan88 committed:
Remove .idea folder from repository

pyclan88 committed:
Update files and remove .idea from tracking
